src/v2/Components/ArtworkFilter/ArtworkFilters/SizeFilter.tsx

```diff
@@ -1,65 +1,212 @@
+    height: parseRange(height),
+    width: parseRange(width),
+  }
+
+  // @ts-expect-error STRICT_NULL_CHECK
+  const [showCustom, setShowCustom] = useState(!!hasValue(initialCustomSize))
+  const [customSize, setCustomSize] = useState<CustomSize>(
+    // @ts-expect-error STRICT_NULL_CHECK
+    hasValue(initialCustomSize) ? initialCustomSize : DEFAULT_CUSTOM_SIZE
+  )
+  const [mode, setMode] = useState<"resting" | "done">("resting")
+
+  const handleInputChange = (dimension: "height" | "width", index: number) => ({
+    currentTarget: { value },
+  }: React.FormEvent<HTMLInputElement>) => {
+    const isOpenEnded = value === "" || value === "0"
+    const isMin = index === 0
 
-  const toggleSelection = (selected, name) => {
+    setCustomSize(prevCustomSize => {
+      if (isOpenEnded) {
+        prevCustomSize[dimension][index] = "*"
+      } else {
+        const parsedValue = parseInt(value, 10)
+        if (prevCustomSize[dimension])
+          prevCustomSize[dimension][index] = parsedValue
+        else if (isMin) {
+          prevCustomSize[dimension] = [parsedValue, "*"]
+        } else {
+          prevCustomSize[dimension] = ["*", parsedValue]
+        }
+      }
+
+      return { ...prevCustomSize }
+    })
+  }
+
+  const toggleSizeSelection = (selected: boolean, name: string) => {
     // @ts-expect-error STRICT_NULL_CHECK
-    let sizes = filterContext.currentlySelectedFilters().sizes.slice()
+    let sizes = currentlySelectedFilters().sizes.slice()
     if (selected) {
       sizes.push(name)
     } else {
       sizes = sizes.filter(item => item !== name)
     }
-    filterContext.setFilter("sizes", sizes)
+
+    const newFilters = {
+      // @ts-expect-error STRICT_NULL_CHECK
+      ...currentlySelectedFilters(),
+      sizes,
+      height: "*-*",
+      width: "*-*",
+    }
+
+    // @ts-expect-error STRICT_NULL_CHECK
+    setFilters(newFilters, { force: false })
+    setCustomSize({ height: ["*", "*"], width: ["*", "*"] })
+  }
+
+  const handleClick = () => {
+    const newFilters = {
+      // @ts-expect-error STRICT_NULL_CHECK
+      ...currentlySelectedFilters(),
+      sizes: [],
+      // @ts-expect-error STRICT_NULL_CHECK
+      ...mapSizeToRange(convertSizeToInches(customSize)),
+    }
+    // @ts-expect-error STRICT_NULL_CHECK
+    setFilters(newFilters, { force: false })
+    setMode("done")
   }
 
   const tokens = useThemeConfig({
     v2: { my: 0.5, secondaryVariant: "small" as TextVariant },
     v3: { my: 1, secondaryVariant: "xs" as TextVariant },
   })
 
+  useEffect(() => {
+    // if filter state is being reset, then also clear local input state
+    if (reset) {
+      setCustomSize({ height: ["*", "*"], width: ["*", "*"] })
+    }
+  }, [reset])
+
+  // @ts-expect-error STRICT_NULL_CHECK
+  const selection = currentlySelectedFilters().sizes
+  // @ts-expect-error STRICT_NULL_CHECK
+  const customHeight = currentlySelectedFilters().height
+  // @ts-expect-error STRICT_NULL_CHECK
```

Comment:
We'll want to avoid adding these `@ts-expect-error STRICT_NULL_CHECK` comments in favor of making sure things are type-safe. Would something like 
```
currentlySelectedFilters()?.width
```
fix it? 

Another way to do it is:

```tsx
const selectedFilters = currentlySelectedFilters()! 
```

of which the `!` says: we're certain that `currentlySelectedFilters()` returns a non-null value. 

If it is possible that `currentlySelectedFilters()` can be null, we'll want to handle that properly however.

```tsx
const selectedFilters = currentlySelectedFilters()

if (selectedFilters) {
  ... do stuff 
}
```

Reply:
Thank you, that's very helpful!